mantidproject/mantid
# SaveASCII - one spectrum per file

**Description of work.**

This PR introduces a new boolean parameter for the SaveASCII algorithm: `OneSpectrumPerFile`. 
When the parameter is set to `true`, instead of concatenating all the histograms into one single ascii file, the algorithm will produce one file per spectrum.

**To test:**

* Create a sample workspace
```python
from mantid.simpleapi import *
CreateSampleWorkspace(OutputWorkspace="ws")
```
* Export with `SaveASCII`, use the `OneSpectrumPerFile` option to modify the behavior.
* Observe the files

Part of #31943 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
